tests/unit/sagemaker/workflow/test_parameters.py

```diff
@@ -35,65 +35,84 @@ def test_parameter():
     }
+    assert param.expr == {"Get": "Parameters.MyString"}
+    assert param.parameter_type.python_type == str
 
 
 def test_parameter_with_invalid_default():
     with pytest.raises(TypeError):
         ParameterFloat(name="MyFloat", default_value="abc")
 
 
-def test_parameter_string_implicit_value():
-    param = ParameterString("MyString")
-    assert param.__str__() == ""
-    param1 = ParameterString("MyString", "1")
-    assert param1.__str__() == "1"
-    param2 = ParameterString("MyString", default_value="2")
-    assert param2.__str__() == "2"
-    param3 = ParameterString(name="MyString", default_value="3")
-    assert param3.__str__() == "3"
-    param3 = ParameterString(name="MyString", default_value="3", enum_values=["3"])
-    assert param3.__str__() == "3"
+def test_parameter_to_string_and_string_implicit_value():
+    param = ParameterString("MyString", "1")
+
+    assert param.to_string() == param
+
+    with pytest.raises(TypeError) as error:
+        str(param)
+
+    assert str(error.value) == "Pipeline variables do not support __str__ operation."
 
 
 def test_parameter_integer_implicit_value():
-    param = ParameterInteger("MyInteger")
-    assert param.__int__() == 0
-    param1 = ParameterInteger("MyInteger", 1)
-    assert param1.__int__() == 1
-    param2 = ParameterInteger("MyInteger", default_value=2)
-    assert param2.__int__() == 2
-    param3 = ParameterInteger(name="MyInteger", default_value=3)
-    assert param3.__int__() == 3
+    param = ParameterInteger("MyInteger", 1)
+
+    with pytest.raises(TypeError) as error:
+        int(param)
+
+    assert str(error.value) == "Pipeline variables do not support __int__ operation."
 
 
 def test_parameter_float_implicit_value():
-    param = ParameterFloat("MyFloat")
-    assert param.__float__() == 0.0
-    param1 = ParameterFloat("MyFloat", 1.1)
-    assert param1.__float__() == 1.1
-    param2 = ParameterFloat("MyFloat", default_value=2.1)
-    assert param2.__float__() == 2.1
-    param3 = ParameterFloat(name="MyFloat", default_value=3.1)
-    assert param3.__float__() == 3.1
+    param = ParameterFloat("MyFloat", 1.1)
+
+    with pytest.raises(TypeError) as error:
+        float(param)
+
+    assert str(error.value) == "Pipeline variables do not support __float__ operation."
 
 
 def test_parsable_parameter_string():
     param = ParameterString("MyString", default_value="s3://foo/bar/baz.csv")
     assert urlparse(param).scheme == "s3"
+
+
+def test_startswith_on_parameter_string():
+    param = ParameterString("MyString", default_value="s3://foo/bar/baz.csv")
+    # The param will only be parsed in runtime (Pipeline backend) so not able to tell in SDK
+    assert not param.startswith("s3")
+
+
+def test_add_func():
+    param_str = ParameterString(name="MyString", default_value="s3://foo/bar/baz.csv")
+    param_int = ParameterInteger(name="MyInteger", default_value=3)
+
+    with pytest.raises(TypeError) as error:
+        param_str + param_int
```

Comment:
shall we cover all combinations (tips: `pytest.mark.parameterized`)?

Reply:
Sure, will add

Reply:
The `@pytest.mark.parametrize` will call `__str__` which cases all arguments to string, which operation is not allowed in `PipelineVarialble`. 
Therefore, I have to manually add more test cases to cover. 
Plus considering the `__add__` is defined on the top in the base `PipelineVariable` class, we may not need to iterating all combinations as there're so many (not just the parameters, but also include other vars like properties, Join funcs etc.). A few test sample can be good enough.